src/command/history.rs

```diff
@@ -49,16 +49,24 @@ impl<'a, S> BaseCommand for HistoryCommand<'a, S> {
     }
 
     fn execute(&self, shell: &mut Shell<S>, _: &[String]) -> Result<String> {
-        // A bit of a mouthful. We grab the underlying history of the shell, iterate it, map the
-        // elements to strings, then collection them into a vector of Strings before we join them
-        // with a newline + tab.
-        let history_output = shell
-            .rl
-            .history()
-            .iter()
-            .map(|h| h.to_string())
-            .collect::<Vec<String>>()
-            .join("\n\t");
+        // A bit of a mouthful. We grab the underlying history of the shell, collect its elements
```

Comment:
Regarding this change, I'm assuming that the API changed and `rl.history()` is no longer an iterator? If not, what's the rationale for this change? To be honest, I think I prefer it anyways, after having written more Rust since I first wrote `shi`, I have gotten far less interested in trying to concoct complex iterator chains 😅.